tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go

```diff
@@ -892,6 +892,11 @@ func computeIterations(run *v1alpha1.Run, tls *pipelineloopv1alpha1.PipelineLoop
 			}
 		}
 	}
+	if from == to && to == 0 && step != 0 {
```

Comment:
I think the use case is when from == to, then the iteration is equals to 1. To doesn't have to be 0. 

Another edge case could be that `to` - `from` < `step` if step > 0 and `to` - `from` > `step` for step < 0, then the iteration is also 1.